pkg/scheduler/framework/plugins/podtopologyspread/scoring.go

```diff
@@ -149,9 +149,12 @@ func (pl *PodTopologySpread) PreScore(
 			return
 		}
 		// (1) `node` should satisfy incoming pod's NodeSelector/NodeAffinity
-		// (2) All topologyKeys need to be present in `node`
+		// (2) all topologyKeys need to be present in `node`
+		// (3) `node` doesn't have `node.kubernetes.io/unschedulable:NoSchedule` taint or `node` has this taint but pod tolerates it
 		match, _ := requiredNodeAffinity.Match(node)
-		if !match || (requireAllTopologies && !nodeLabelsMatchSpreadConstraints(node.Labels, state.Constraints)) {
+		if !match ||
+			(requireAllTopologies && !nodeLabelsMatchSpreadConstraints(node.Labels, state.Constraints)) ||
+			(matchUnSchedulableTaint(node.Spec.Taints) && !matchUnSchedulableTaintToleration(pod.Spec.Tolerations)) {
```

Comment:
This is not necessary as the evaluating node should have already passed all taint/toleration constraints.

(please also tweak or eliminate the delta UTs)